cpp/include/cudf/strings/detail/convert/fixed_point.cuh

```diff
@@ -127,6 +133,11 @@ __device__ thrust::optional<int32_t> parse_exponent(char const* iter, char const
 template <typename DecimalType>
 __device__ DecimalType parse_decimal(char const* iter, char const* iter_end, int32_t scale)
 {
+  if (iter_end == iter) {
+     // No digits found
+    return DecimalType{0};
+  }
```

Comment:
This if-statement is also not be necessary. The current code logic handles empty strings already.